handlers/telegram/telegram.go

```diff
@@ -188,29 +188,33 @@ func (h *handler) SendMsg(ctx context.Context, msg courier.Msg) (courier.MsgStat
 
 	// figure out whether we have a keyboard to send as well
 	qrs := msg.QuickReplies()
-	var keyboard *ReplyKeyboardMarkup
-	if len(qrs) > 0 {
-		keyboard = NewKeyboardFromReplies(qrs)
-	}
 
 	// if we have text, send that if we aren't sending it as a caption
 	if msg.Text() != "" && caption == "" {
+		var msgKeyBoard *ReplyKeyboardMarkup
+		if len(msg.Attachments()) == 0 && len(qrs) > 0 {
+			msgKeyBoard = NewKeyboardFromReplies(qrs)
+		}
+
 		form := url.Values{
 			"chat_id": []string{msg.URN().Path()},
 			"text":    []string{msg.Text()},
 		}
 
-		externalID, log, err := h.sendMsgPart(msg, authToken, "sendMessage", form, keyboard)
+		externalID, log, err := h.sendMsgPart(msg, authToken, "sendMessage", form, msgKeyBoard)
 		status.SetExternalID(externalID)
 		hasError = err != nil
 		status.AddLog(log)
 
-		// clear our keyboard which has now been sent
-		keyboard = nil
```

Comment:
I'm not getting why this doesn't work as is..

Reply:
We are setting the keyboard on the first message instead of the last one so the last one is having `"remove_keyboard":true`

Reply:
The change is to make sure we always set the keyboard only on the last one

Reply:
Ah ok - it needs to be on the last rather than the first! That make sense. 

Maybe cleaner to still create one keyboard instance at the top if `len(qrs) > 0`

Reply:
Done